pyop2/codegen/rep2loopy.py

```diff
@@ -288,8 +293,8 @@ def runtime_indices(expressions):
     for node in traversal(expressions):
         if isinstance(node, RuntimeIndex):
             indices.append(node.name)
-
-    return frozenset(indices)
+    # use a dict as an ordered set
+    return {i: None for i in indices}
```

Comment:
This ensures ordering but also changes the return type.
Why not:
```python
return {i: None for i in indices}.keys()
```
Or
```python
return type(indices)(dict.fromkeys(indices))
```

Reply:
I need to return a type that can act as an ordered set. The return type needs to be changed because I continue to need ordered set semantics outside of the function.

Reply:
Ok, that's fine then.